controllers/spec/common.go

```diff
@@ -204,6 +204,19 @@ func getDownloadCommand(downloadPath, componentPackage string, authProvided, tls
 		"--admin-url",
 		"$webServiceURL",
 	}
+	if tlsProvided {
+		args = []string{
+			"if [ \"b$tlsAllowInsecureConnection\" = \"btrue\" ]; then export ALLOW_INSECURE=\"--tls-allow-insecure\"; fi && ",
+			"if [ \"b$tlsHostnameVerificationEnable\" = \"btrue\" ]; then export ENABLE_HOSTNAME_VERIFY=\"--tls-enable-hostname-verification\"; fi && ",
```

Comment:
could you please check https://github.com/apache/pulsar/blob/master/docker/pulsar/scripts/apply-config-from-env.py, which is contained in each runner image by default, and it should be useful to overwrite the tls / auth configs in `conf/client.conf` from ENV, and can be used to simplify the `pulsar-admin` command here.

Reply:
@jiangpengcheng I think it is acceptable, but it requires some work to keep the backward compatibility.

Reply:
for compatibility, we may need to use shell to check whether `python` is installed on the runtime image which makes the command more complicated, any other advice for compatibility?

Reply:
@freeznet 
looks like we can use `env > conf/client.conf` to do similar thing, but some config names we used in configs and secrets need to be transformed when writing into the config file:

```
webServiceURL -> webServiceUrl
brokerServiceURL -> brokerServiceUrl
clientAuthenticationPlugin -> authPlugin
clientAuthenticationParameters -> authParams
tlsHostnameVerificationEnable -> tlsEnableHostnameVerification
```

for this transformation, I think we can inform users to update their configs&secrets before upgrading function-mesh, HDYT?

Reply:
@jiangpengcheng `env > conf/client.conf` needs extra commands as well right? 
What do you think if we separate the package download part as an `initContainer`? will it solve the backward compatibility issue?

Reply:
sounds perfect! but it will be a big change(needs to add an init container and volume if download path is specified?), and we still need to assemble download commands in the init container

Reply:
Correct, the file path can be extracted via `.spec.java.jar` / `.spec.python.py` / `.spec.golang.go`.

Reply:
hm, during development, I found that `.spec.java.jar` can be any value, like a raw filename without specifying dir, like `pulsar-functions-api-examples.jar`, then in the `VolumeMounts` for downloader, we need to use `/pulsar`(the workdir) as the `MountPath`? which will override everything under the workdir